cfpb/consumerfinance.gov
# Upgrade to Wagtail 2.12.3

In Wagtail 2.12 stream_data on StreamField values is deprecated

https://docs.wagtail.io/en/stable/releases/2.12.html


## Additions

- Add raw_data to export_mega_menu management command
- Add raw_data to import_mega_menu management command
- @willbarton worked on a fix for the wagtailembeds migration failure issue

## Changes

I took the approach of using `raw_data` as `steam_data` is deprecated. In `cfgov/v1/util/migrations.py`, `get_streamfield_data` and `set_streamfield_data` are modified to use `raw_data`.

As a result of the above change I fixed a TypeError: Object of type 'RawDataView' is not JSON serializable


## How to test this PR

- Verify there are no failures running wagtail migrations and check generated indexes

1. tox -e unittest-future
2. python cfgov/manage.py migrate
3. python cfgov/manage.py runscript export_enforcement_actions
4. python cfgov/manage.py runscript update_oaa_status
5. python cfgov/manage.py runscript export_ask_data
6. python cfgov/manage.py runscript fix_links_in_tableblocks
7. python cfgov/manage.py update_chart_block_dates --snapshot_file

1. cfgov/manage.py export_mega_menu en output.json
2. cfgov/manage.py export_mega_menu es output.json OR
3. cfgov/manage.py export_mega_menu en (writes to stdout)

4. cfgov/manage.py import_mega_menu en import.json
5. cfgov/manage.py import_mega_menu es import.json OR
6. cfgov/manage.py import_mega_menu en (reads from stdin) 

cat new.json | cfgov/manage.py import_mega_menu en

Current menu:

curl -k https://github.local/raw/gist/chosak/4600af0d52fbe588a634ea6fee54f942/raw/5af1c1428f63ddfd7f6fc8ba9
fa115b0ac17235a/current.json | cfgov/manage.py import_mega_menu en

New menu:

curl -k https://github.local/raw/gist/chosak/fdeab64f382ca0d17568cdbd717862d8/raw/9b2e8fb73062ec631c609d69c1e9ee49261b6459/new.json | cfgov/manage.py import_mega_menu en

After importing a new mega menu JSON file, verify changes to the mega menu are visible in http://localhost:8000/


## Notes and todos

- Send email notification before merging this PR


## Checklist

  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

#### Browser testing

Visually tested in the following supported browsers:

#### Other
